chainer/training/extensions/linear_shift.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import division
+
+from chainer.training import extension
+
+
+class LinearShift(extension.Extension):
+
+    """Trainer extension to change an optimizer attribute linearly.
+
+    This extension changes an optimizer attribute from the first value to the
+    last value linearly within a specified duration. The typical use case is
+    warming up of the momentum coefficient.
+
+    For example, suppose that this extension is called at every iteration, and
+    ``value_range == (x, y)`` and ``time_range == (i, j)``. Then, this
+    extension keeps the attribute to be ``x`` up to the ``i``-th iteration, and
+    linearly shifts the value to ``y`` by the ``j``-th iteration.
```

Comment:
Strictly speaking, if the trigger is not default one (i.e. `1, iteration`), `_t` attribute and the number of iterations is not identical. So it is better to add the description like "if the trigger is the default one".